presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -804,11 +814,153 @@ public void testSchemaEvolution()
         assertUpdate("DROP TABLE test_schema_evolution");
     }
 
+    @Test
+    public void testRcAndTextFormats()
+            throws IOException
+    {
+        getQueryRunner().execute("CREATE TABLE lineitem_ex_partitioned WITH (format='ORC', partitioned_by=ARRAY['ds']) AS " +
+                "SELECT linenumber, orderkey, partkey, suppkey, quantity, extendedprice, tax, shipinstruct, shipmode, ship_by_air, " +
+                "is_returned, ship_day, ship_month, ship_timestamp, commit_timestamp, discount_real, discount, tax_real, ship_day_month, " +
+                "discount_long_decimal, tax_short_decimal, long_decimals, keys, doubles, nested_keys, flags, reals, info, dates, timestamps, " +
+                "comment, uppercase_comment, fixed_comment, char_array, varchar_array, '2019-11-01' AS ds FROM lineitem_ex");
+        try {
+            for (HiveStorageFormat format : ImmutableList.of(RCBINARY, RCTEXT, TEXTFILE)) {
+                assertFileFormat(format);
+            }
+        }
+        finally {
+            assertUpdate("DROP TABLE lineitem_ex_partitioned");
+        }
+    }
+
+    private void assertFileFormat(HiveStorageFormat storageFormat)
+            throws IOException
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        try {
+            queryRunner.execute("CREATE TABLE test_file_format_orc WITH (format='ORC', partitioned_by=ARRAY['ds']) AS " +
```

Comment:
just like in the other place, CREATE TABLE needs to come before the try block. Otherwise, failure in CREATE TABLE will result in a failure in DROP TABLE and the final error (table not exists) won't be useful for troubleshooting.